http/server.ts

```diff
@@ -371,21 +371,23 @@ export class Server {
           error instanceof Deno.errors.ConnectionReset ||
           error instanceof Deno.errors.NotConnected
         ) {
-          // Backoff after transient errors to allow time for the system to
-          // recover, and avoid blocking up the event loop with a continuously
-          // running loop.
-          if (!acceptBackoffDelay) {
-            acceptBackoffDelay = INITIAL_ACCEPT_BACKOFF_DELAY;
-          } else {
-            acceptBackoffDelay *= 2;
+          if (!this.#closed) {
+            // Backoff after transient errors to allow time for the system to
+            // recover, and avoid blocking up the event loop with a continuously
+            // running loop.
+            if (!acceptBackoffDelay) {
+              acceptBackoffDelay = INITIAL_ACCEPT_BACKOFF_DELAY;
+            } else {
+              acceptBackoffDelay *= 2;
+            }
+
+            if (acceptBackoffDelay >= MAX_ACCEPT_BACKOFF_DELAY) {
+              acceptBackoffDelay = MAX_ACCEPT_BACKOFF_DELAY;
+            }
+
+            await delay(acceptBackoffDelay);
```

Comment:
Isn't the following sequence still possible?

1. Server is accepting traffic.
2. Server enters `delay()`.
3. Server is closed while `delay()` is still executing.
4. Test runner reports leaked op.

Reply:
Agreed, this should be solvable as well by aborting the delay when closing the server. I gave it a go [here](https://github.com/denoland/deno_std/pull/2732/commits/5da431fc42bfb95966fa4afd7bb4a55f1d2bf5f5). However, I think there is room to improve the test case to not have it depend on the leaking detection AssertionError.